hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/AbstractYarnScheduler.java

```diff
@@ -757,10 +757,9 @@ private void completeOustandingUpdatesWhichAreReserved(
       RMContainer rmContainer, ContainerStatus containerStatus,
       RMContainerEventType event) {
     N schedulerNode = getSchedulerNode(rmContainer.getNodeId());
-    if (schedulerNode != null &&
-        schedulerNode.getReservedContainer() != null) {
+    if (schedulerNode != null) {
       RMContainer resContainer = schedulerNode.getReservedContainer();
-      if (resContainer.getReservedSchedulerKey() != null) {
+      if (resContainer != null && resContainer.getReservedSchedulerKey() != null) {
```

Comment:
It is recommended to add a try catch module. I have made this change in the production environment before, but it still reports a null pointer.

Reply:
Thanks @zeekling for the review.
I'm not sure why your environment still reported NPE after changing like that, `resContainer.getReservedSchedulerKey()` won't throw NPE any more if resContainer is null. Could you please attach some details of the NPE and your changes?
For this change,  I think NPE should be fixed instead of be caught in general.

Reply:
I modified like your change.

Reply:
I can't find any chance to reproduce the NPE, could you please explain in which case NPE can be thrown again?  If we can locate that risk point, I would like to fix it instead of catching it in try block. Thanks.